mlflow/server/js/src/common/utils/AuthComponent.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import Utils from '../../common/utils/Utils';
+import { Redirect } from 'react-router';
+import PropTypes from 'prop-types';
+
+export class AuthComponentImpl extends Component {
+  constructor(props) {
+    super(props);
+    const urlState = Utils.getSearchParamsFromUrl(props.location.search);
+    this.state = {
+      persistedState: {
+        code: urlState.code === undefined ? '' : urlState.code,
+      },
+    };
+  }
+
+  static propTypes = {
+    location: PropTypes.object,
+  };
+
+  componentDidMount = () => {
+    const req = new XMLHttpRequest();
+    req.open('GET', '/token?code=' + this.state.persistedState.code, false);
```

Comment:
Here in the angular lib, we pass back the redirect URL (I known that it is possible to "compute" in the backend), but maybe it is better to do the same for coherency reason (it is recomputed in the front like: window.location.protocol + '//' + window.location.host + this.location.prepareExternalUrl(this.redirectUriRoute))

Reply:
This is what I do to compute the redirect uri I pass to the sso, but here is the call made to the backend to get the token, should I do something similar ? I am not sure it is needed since the  window.location.protocol + '//' + window.location.host is automatic when using the XHR object, and the /token is the backend endpoint I need to call.